src/goto-checker/solver_factory.cpp

```diff
@@ -178,12 +178,16 @@ static std::unique_ptr<SatcheckT>
 make_satcheck_prop(message_handlert &message_handler, const optionst &options)
 {
   auto satcheck = util_make_unique<SatcheckT>(message_handler);
-  if(options.is_set("write-solver-stats-to"))
+  if(auto hardness_collector = dynamic_cast<hardness_collectort *>(&*satcheck))
```

Comment:
Hm, the idea was that satchecks that don't want to do hardness collection could just implement these functions as no-ops?

Reply:
Then you should have implemented it in all existing satchecks, which are broken now.